### PR TITLE
Add a Test Case Guide adoc

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

File: test-case-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+== Annotations
+
+When using "test templates" you can annotate a  single test or a whole test class with one of the following  annotations:
+
+* FailureExpected - allows to skip a single test or all test of a class, because test failures are expected. The test will acutally run, but not lead to an error report. In fact if a test is marked with `@FailureExpected` and the test actually succeed an error occurs. As a parameters to this annotation a jira key is required.
+* RequiresDialect - tests methods/classes annotated with `@RequiresDialect` will only run if the current Dialect is matching the one specified on as annotation parameter. You can also specify a comment and/or jira key explaining why this test requires a certain dialect
+* RequiresDialectFeature - tests methods/classes annotated with `@RequiresDialectFeature` will only run if the current Dialect offers the specified feature. Examples for this features are `SupportsSequences`, `SupportsExpectedLobUsagePattern` or `SupportsIdentityColumn`s. You can add more feature if you need to. Have a look at `DialectChecks`.
```
Comment (NathanQingyangXu) [inline test-case-guide.adoc]: From my previous User Guide improvement endeavour, ASCIIDoc has bug rendering \`xxx\`s. I bypassed the issue by \`\`xxx\`\`s.

Comment (dreab8) [reply test-case-guide.adoc]: Thanks @NathanQingyangXu, my error it should be `SupportsIdentityColumns`. Just push a commit with a fix

**Comment by dreab8** [comment]
thanks @sebersole

**Comment by sebersole** [comment]
Good catch @NathanQingyangXu

**Pull request merged**